panel/widgets/tables.py

```diff
@@ -371,6 +371,7 @@ def _sort_df(self, df: pd.DataFrame) -> pd.DataFrame:
 
         # Temporarily add _index_ column because Tabulator uses internal _index
         # as additional sorter to break ties
+        df = df.copy()
```

Comment:
I want to flag this line. Before this change, we changed the column dtype of the original dataframe. @maximlt I know you have spent some time sorting with tabulator/python.

Reply:
Could you precise what warning this is fixing and which column dtype is changed?

Reply:
The following warnings, in the UI test ([example](https://github.com/holoviz/panel/actions/runs/9704627234/job/26785216150#step:6:150))
``` python-traceback
panel/tests/ui/widgets/test_tabulator.py: 27 warnings
  D:\a\panel\panel\panel\widgets\tables.py:374: SettingWithCopyWarning: 
  A value is trying to be set on a copy of a slice from a DataFrame.
  Try using .loc[row_indexer,col_indexer] = value instead
  
  See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
    df['_index_'] = np.arange(len(df)).astype(str)

panel/tests/ui/widgets/test_tabulator.py: 27 warnings
  D:\a\panel\panel\panel\widgets\tables.py:403: SettingWithCopyWarning: 
  A value is trying to be set on a copy of a slice from a DataFrame
  
  See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
    df.drop(columns=['_index_'], inplace=True)
```

And it change the dtype of the column itself (`df.columns.dtype`), see `test_tabulator_sorters_unnamed_index` and `test_tabulator_sorters_int_name_column`

Reply:
Ok thanks. I think the warning is caused by the fact that `_sort_df` is called after `_filter_dataframe` which may return a view of `self.value`. If you think that's the reason, adding a comment wouldn't hurt (and there's maybe a better place where to call `.copy()`).